lib/table_of_contents/parser.rb

```diff
@@ -45,6 +48,9 @@ def parse_content
                .downcase
                .gsub(PUNCTUATION_REGEXP, '') # remove punctuation
                .tr(' ', '-') # replace spaces with dash
+          if (@configuration.anchor_id_url_encoded)
+            id = url_encode(id)
```

Comment:
I'd like you to use `CGI.escape` to encode string for URL.

```suggestion
            id = CGI.escape(id)
```

The result is same:

```
irb(main):014:0> url_encode('あああ')
=> "%E3%81%82%E3%81%82%E3%81%82"
irb(main):015:0> CGI.escape('あああ')
=> "%E3%81%82%E3%81%82%E3%81%82"
```